catalog/core/catalog-core-commands/src/main/java/org/codice/ddf/commands/catalog/ValidateCommand.java

```diff
@@ -0,0 +1,157 @@
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.commands.catalog;
+
+import static org.apache.karaf.shell.support.ansi.SimpleAnsi.COLOR_CYAN;
+import static org.apache.karaf.shell.support.ansi.SimpleAnsi.COLOR_DEFAULT;
+import static org.apache.karaf.shell.support.ansi.SimpleAnsi.COLOR_RED;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.Iterator;
+import java.util.List;
+
+import org.apache.commons.io.IOUtils;
+import org.apache.karaf.shell.api.action.Action;
+import org.apache.karaf.shell.api.action.Argument;
+import org.apache.karaf.shell.api.action.Command;
+import org.apache.karaf.shell.api.action.Completion;
+import org.apache.karaf.shell.api.action.lifecycle.Reference;
+import org.apache.karaf.shell.api.action.lifecycle.Service;
+import org.apache.karaf.shell.support.completers.FileCompleter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.catalog.data.Metacard;
+import ddf.catalog.data.impl.MetacardImpl;
+import ddf.catalog.util.Describable;
+import ddf.catalog.validation.MetacardValidator;
+import ddf.catalog.validation.ValidationException;
+
+@Command(scope = CatalogCommands.NAMESPACE, name = "validate", description = "Run all pertinent validators against the specified metadata file")
+@Service
+public class ValidateCommand implements Action {
+
+    @Argument(index = 0, name = "fileName", description = "The path to the file that you want to validate", required = true, multiValued = false)
+    @Completion(FileCompleter.class)
+    String filename;
+
+    @Reference
+    List<MetacardValidator> validators;
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ValidateCommand.class);
+
+    protected PrintStream console = System.out;
+
+    private static final String INDENT =       "|   ";
+    private static final String LAST_INDENT =  "    ";
+    private static final String ELEMENT =      "+-- ";
+    private static final String LAST_ELEMENT = "`-- ";
+
+    private static final String WARNING_TITLE = "Warnings";
+    private static final String ERROR_TITLE =   "Errors";
+
+    private static final String WARNING_COLOR = COLOR_CYAN;
+    private static final String ERROR_COLOR =   COLOR_RED;
+
+    public Object execute() throws Exception {
+
+        File fileToValidate = new File(filename);
+        if (!fileToValidate.exists()) {
+            console.printf("Unable to locate file %s. Double check the path is correct and the file exists.%n", filename);
+            return null;
+        }
+
+        Metacard metacard;
+        try {
+            metacard = createMetacard();
+        } catch (IOException e) {
+            console.println(ERROR_COLOR + "Error reading file. Check the log for the stacktrace" + COLOR_DEFAULT);
+            LOGGER.error("Error trying to read file {}", filename, e);
+            return null;
+        }
+
+        Iterator<MetacardValidator> iterator = validators.iterator();
```

Comment:
I wanted to model the output after the 'tree' command in linux because I thought it would make the output easier tor read. Thus I format the string differently if it's the last element in the list. Using an iterator was the cleanest way I could think to achieve that.